sdcm/nemesis.py

```diff
@@ -2158,8 +2181,16 @@ def delete_half_partition(self, ks_cf):
             return partitions_for_delete
 
         queries = []
+        partition_percentage = min(max(random.random(), 0.25), 0.75)
         for pkey, ckey in partitions_for_delete.items():
-            queries.append(f"delete from {ks_cf} where pk = {pkey} and ck > {int(ckey[1] / 2)}")
+            if random.random() < 0.5:
+                queries.append(f"delete from {ks_cf} where pk = {pkey} and ck > {int(ckey[1] / 2)}")
+            else:
+                self.log.debug("Using USING TIMESTAMP clause in the deletion for this partition: %s", pkey)
+                timestamp = self.get_random_timestamp_from_partition(
+                    ks_cf=ks_cf, pkey=pkey, partition_percentage=partition_percentage)
+                queries.append(f"delete from {ks_cf} using timestamp {timestamp} where pk = {pkey}")
+
```

Comment:
this is inside `delete_half_partition` function which by definition deletes half partition... which now is not true.
Let's move it somewhere else and do the decision which one to ouse earlier so it's clearer what was the flow of the test.

Reply:
I've moved the code (along with other required bits) into a separate `delete_by_range_using_timestamp` method and now the decision which path to take happens in the disruption func `disrupt_delete_by_rows_range`.